src/services/terms.ts

```diff
@@ -32,7 +32,7 @@ export class TermsTransformer {
 
   fromQuad(quad: RDF.Quad): void {
     const subject = quad.subject;
-    if (!subject.equals(this.currentTerm.id)) {
+    if (this.currentTerm.id === undefined || !subject.equals(this.currentTerm.id)) {
```

Comment:
The new condition isn't really necessary: if `this.currentTerm.id` is undefined, `subject.equals()` returns true?

Reply:
This caused `gts check` violation before.

Reply:
Just

```
if (!subject.equals(this.currentTerm.id)) {
```

gives:

```
Error:(35, 25) TS2345: Argument of type 'Term | undefined' is not assignable to parameter of type 'Term'.
  Type 'undefined' is not assignable to type 'Term'.
```

Reply:
Good one! :)

(Somewhat strange, though - the [RDFJS spec](https://rdf.js.org/data-model-spec/#term-interface) states that `equals()` returns `true` if (among others) the parameter is not `undefined`, so I expected `undefined` to be a valid value.)